File: 000_C# Training/C#_9 Lists/Java/MainClass.java
import java.util.ArrayList;
import java.util.Scanner;

class MainClass
{
    public static void main(String[] args)
    {
        CreateAndFillList();

        ArrayList<Integer> myIntList = CreateAndFillListAndReturnIt();

        CheckIfItemExists();

        LoopThroughAllElementsOfList(myIntList);
        
        ArrayList<Integer[]> myNestedIntList = CreateAndFillNestedList();

        LoopThroughAllElementsOfNestedList(myNestedIntList);

        Exercise_EliminateDoubleAppearances();

        Pitfall_1();
        
        new Scanner(System.in).nextLine();
    }

    public static void CreateAndFillList()
    {
        ArrayList<Integer> myIntList = new ArrayList<Integer>();
        ArrayList<String> myStringList = new ArrayList<String>();

        myIntList.add(34);
        myIntList.add(34);

        for (int i=5;i<8;i++)
            myIntList.add(i);

        myStringList.add("34");
        myStringList.add("34");

        for (int i=5;i<8;i++)
            myStringList.add(Integer.toString(i));
    }

    public static ArrayList<Integer> CreateAndFillListAndReturnIt()
    {
        ArrayList<Integer> myIntList = new ArrayList<Integer>();

        myIntList.add(34);
        myIntList.add(34);

        for (int i=5;i<8;i++)
            myIntList.add(i);

        return myIntList;
    }

    public static void CheckIfItemExists()
    {
        ArrayList<Integer> myIntList = new ArrayList<Integer>();

        if (!myIntList.contains(45))
            myIntList.add(45);
    }

    public static void LoopThroughAllElementsOfList(ArrayList<Integer> myIntList)
    {
        for (int item : myIntList)
            System.out.println(item);

        for (int i=0;i<myIntList.size();i++)
            System.out.println(myIntList.get(i));
    }

    public static ArrayList<Integer[]> CreateAndFillNestedList()
    {
        ArrayList<Integer[]> myIntList = new ArrayList<Integer[]>();

        Integer[] tempInt;
        int testCounter = 0;

        for (int i=5;i<8;i++)
        {
            tempInt = new Integer[10];

            for (int j=0;j<10;j++)
            {
                ++testCounter;
                tempInt[j] = testCounter;
            }

            myIntList.add(tempInt);
        }

        return myIntList;
    }

    public static void LoopThroughAllElementsOfNestedList(ArrayList<Integer[]> myNestedIntList)
    {
        for (Integer[] itemArray : myNestedIntList)
            for (int item : itemArray)
                System.out.println(item);

        for (int j=0;j<myNestedIntList.size();j++)
            for (int i=0;i<myNestedIntList.get(j).length;i++)
                System.out.println(myNestedIntList.get(j)[i]);
    }

    public static void Exercise_EliminateDoubleAppearances()
    {
        Integer[] inputNumbers = new Integer[] {3, 7, 5, 5, 4, 3, 2, 7};
        ArrayList<Integer> outputList = new ArrayList<Integer>();

        for (int number : inputNumbers)
            if (!outputList.contains(number))
                outputList.add(number);

        for (int number : outputList)
            System.out.println(number);

    }

    public static void Pitfall_1()
    {
        ArrayList<Integer> myList = new ArrayList<Integer>();

        myList.add(100);
        myList.add(200);

        //OK
        System.out.println(myList.get(0));
        System.out.println(myList.get(1));

        //This will cause an error
        System.out.println(myList.get(100));
        System.out.println(myList.get(200));
    }

}

File: 000_C# Training/C#_9 Lists/C#/MainClass.cs
using System;
using System.Collections.Generic;

class MainClass
{
    public static void Main()
    {
        CreateAndFillList();

        List<int> myIntList = CreateAndFillListAndReturnIt();

        CheckIfItemExists();

        LoopThroughAllElementsOfList(myIntList);
        
        List<int[]> myNestedIntList = CreateAndFillNestedList();

        LoopThroughAllElementsOfNestedList(myNestedIntList);

        Exercise_EliminateDoubleAppearances();

        Pitfall_1();
        
        Console.ReadLine();
    }

    public static void CreateAndFillList()
    {
        List<int> myIntList = new List<int>();
        List<string> myStringList = new List<string>();

        myIntList.Add(34);
        myIntList.Add(34);

        for (int i=5;i<8;i++)
            myIntList.Add(i);

        
        myStringList.Add("34");
        myStringList.Add("34");

        for (int i=5;i<8;i++)
            myStringList.Add(i.ToString());
    }

    public static List<int> CreateAndFillListAndReturnIt()
    {
        List<int> myIntList = new List<int>();

        myIntList.Add(34);
        myIntList.Add(34);

        for (int i=5;i<8;i++)
            myIntList.Add(i);

        return myIntList;
    }

    public static void CheckIfItemExists()
    {
        List<int> myIntList = new List<int>();

        if (!myIntList.Contains(45))
            myIntList.Add(45);
    }

    public static void LoopThroughAllElementsOfList(List<int> myIntList)
    {
        foreach (int item in myIntList)
            Console.WriteLine(item);

        for (int i=0;i<myIntList.Count;i++)
            Console.WriteLine(myIntList[i]);
    }

    public static List<int[]> CreateAndFillNestedList()
    {
        List<int[]> myIntList = new List<int[]>();

        int[] tempInt;
        int testCounter = 0;

        for (int i=5;i<8;i++)
        {
            tempInt = new int[10];

            for (int j=0;j<10;j++)
            {
                ++testCounter;
                tempInt[j] = testCounter;
            }

            myIntList.Add(tempInt);
        }

        return myIntList;
    }

    public static void LoopThroughAllElementsOfNestedList(List<int[]> myNestedIntList)
    {
        foreach (int[] itemArray in myNestedIntList)
            foreach (int item in itemArray)
                Console.WriteLine(item);

        for (int j=0;j<myNestedIntList.Count;j++)
            for (int i=0;i<myNestedIntList[j].Length;i++)
                Console.WriteLine(myNestedIntList[j][i]);
    }

    public static void Exercise_EliminateDoubleAppearances()
    {
        int[] inputNumbers = new int[] {3, 7, 5, 5, 4, 3, 2, 7};
        List<int> outputList = new List<int>();

        foreach (int number in inputNumbers)
            if (!outputList.Contains(number))
                outputList.Add(number);

        foreach (int number in outputList)
            Console.WriteLine(number);

    }

    public static void Pitfall_1()
    {
        List<int> myList = new List<int>();

        myList.Add(100);
        myList.Add(200);

        //OK
        Console.WriteLine(myList[0]);
        Console.WriteLine(myList[1]);

        //This will cause an error
        Console.WriteLine(myList[100]);
        Console.WriteLine(myList[200]);
    }

}